<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="A set of functions for classifying HRSV sequences. It includes
    a shiny app for interactive functionality.">
<title>Molecular Classification of Respiratory Syncytial Virus Sequences ‚Ä¢ resvidex</title>
<script src="deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.jsdelivr.net/gh/afeld/bootstrap-toc@v1.0.1/dist/bootstrap-toc.min.js" integrity="sha256-4veVQbu7//Lk5TSmc7YV48MxtMy98e26cf5MrgZYnwo=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="pkgdown.js"></script><meta property="og:title" content="Molecular Classification of Respiratory Syncytial Virus Sequences">
<meta property="og:description" content="A set of functions for classifying HRSV sequences. It includes
    a shiny app for interactive functionality.">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light"><div class="container">
    
    <a class="navbar-brand me-2" href="index.html">resvidex</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.0.0.9012</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item">
  <a class="nav-link" href="reference/index.html">Reference</a>
</li>
<li class="nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="articles/E1_resvidex_vignette.html">How to use the ReSVidex app</a>
    <a class="dropdown-item" href="articles/E2_resvidex_vignette_R.html">How to use ReSVidex with R</a>
    <a class="dropdown-item" href="articles/E3_resvidex_vignette_results.html">Got the results! What now?</a>
    <a class="dropdown-item" href="articles/E4_an_example.html">Let's get real: a full example with a real dataset</a>
  </div>
</li>
<li class="nav-item">
  <a class="nav-link" href="news/index.html">Changelog</a>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav"></ul>
</div>

    
  </div>
</nav><div class="container template-home">
<div class="row">
  <main id="main" class="col-md-9"><p>Marco Cacciabue, Nahuel Fenoglio, Melina Obreg√≥n</p>
<!-- README.md is generated from README.Rmd. Please edit that file -->
<div class="section level1">
<div class="page-header"><h1 id="resvidex--molecular-classification-of-respiratory-syncytial-virus-sequences">
<strong>ReSVidex</strong>: <img src="reference/figures/hex.png" style="float:right; height:200px;"> Molecular classification of Respiratory Syncytial Virus sequences<a class="anchor" aria-label="anchor" href="#resvidex--molecular-classification-of-respiratory-syncytial-virus-sequences"></a>
</h1></div>
<!-- badges: start -->

<p><a href="https://github.com/marcocacciabue/resvidex/actions/workflows/R-CMD-check.yaml" class="external-link"><img src="https://github.com/marcocacciabue/resvidex/actions/workflows/R-CMD-check.yaml/badge.svg" alt="R-CMD-check"></a> <!-- badges: end --> If you have a working R and Rstudio setup you can install the released version of <strong>ReSVidex</strong> from <a href="https://github.com/" class="external-link">GitHub</a> with:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">require</a></span><span class="op">(</span><span class="st"><a href="https://remotes.r-lib.org" class="external-link">"remotes"</a></span>, quietly <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"remotes"</span><span class="op">)</span></span>
<span>  </span>
<span><span class="fu">remotes</span><span class="fu">::</span><span class="fu"><a href="https://remotes.r-lib.org/reference/install_github.html" class="external-link">install_github</a></span><span class="op">(</span><span class="st">"marcocacciabue/resvidex"</span><span class="op">)</span></span></code></pre></div>
<p>This could take several minutes depending on your system and installation. Only the first time it runs.</p>
<p>Alternatively, you can download the repository as a .zip file and install it manually with Rstudio.</p>
<div class="section level2">
<h2 id="resvidex-is-an-r-package">
<strong>ReSVidex</strong> is an R package<a class="anchor" aria-label="anchor" href="#resvidex-is-an-r-package"></a>
</h2>
<p>This means that users that prefer working directly in an R console can use some of the exported functions. The easiest way is to use the wrapper function ‚Äúclassify()‚Äù</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">#load the library</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">resvidex</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Indicate the file path to the fasta file to use.If your file is in your working directory you need to simply indicate the file name. In this case, we use a test file provided with the package itself. </span></span>
<span></span>
<span><span class="va">file_path</span><span class="op">&lt;-</span><span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>,<span class="st">"test_dataset.fasta"</span>,package<span class="op">=</span><span class="st">"resvidex"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Use the wrapper function. You can change the classification model and pass other arguments</span></span>
<span><span class="fu"><a href="reference/classify.html">classify</a></span><span class="op">(</span>inputFile<span class="op">=</span><span class="va">file_path</span>,model<span class="op">=</span><span class="va">FULL_GENOME</span><span class="op">)</span></span>
<span></span>
<span><span class="co">#If you want to save the results in a file in your working directory, you can run the pipeline and</span></span>
<span><span class="co">#save it as a variable to export</span></span>
<span><span class="va">Classification_Print</span><span class="op">&lt;-</span><span class="fu"><a href="reference/classify.html">classify</a></span><span class="op">(</span>inputFile<span class="op">=</span><span class="va">file_path</span>,model<span class="op">=</span><span class="va">FULL_GENOME</span><span class="op">)</span></span>
<span><span class="fu">utils</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/r/utils/write.table.html" class="external-link">write.csv2</a></span><span class="op">(</span><span class="va">Classification_Print</span>,<span class="st">"Classification_file.csv"</span><span class="op">)</span></span></code></pre></div>
<p>For a more detailed workflow the user can use the exported functions of the package.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># First we indicate the location of the fasta file. In this case, we use a test file provided with the package itself.</span></span>
<span><span class="va">file_path</span><span class="op">&lt;-</span><span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>,<span class="st">"test_dataset.fasta"</span>,package<span class="op">=</span><span class="st">"resvidex"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># We load the sequences</span></span>
<span><span class="va">sequence</span><span class="op">&lt;-</span><span class="fu">ape</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/ape/man/read.dna.html" class="external-link">read.FASTA</a></span><span class="op">(</span><span class="va">file_path</span>,type <span class="op">=</span> <span class="st">"DNA"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># We the count and normalize the k-mers</span></span>
<span><span class="va">NormalizedData</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/kcounter.html">kcounter</a></span><span class="op">(</span>SequenceData<span class="op">=</span><span class="va">sequence</span>,model<span class="op">=</span><span class="va">FULL_GENOME</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># We perform the classification</span></span>
<span><span class="va">PredictedData</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/prediction_caller.html">prediction_caller</a></span><span class="op">(</span>NormalizedData<span class="op">=</span><span class="va">NormalizedData</span>,model<span class="op">=</span><span class="va">FULL_GENOME</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># We add the quality FLAGs</span></span>
<span><span class="va">PredictedData</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/quality_control.html">quality_control</a></span><span class="op">(</span><span class="va">PredictedData</span>,model<span class="op">=</span><span class="va">FULL_GENOME</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># We adjust the classification according to the FLAGs present for each sample:</span></span>
<span><span class="va">PredictedData</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/quality_filter.html">quality_filter</a></span><span class="op">(</span><span class="va">PredictedData</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># The PredictedData dataframe contains the classifications</span></span>
<span></span>
<span><span class="va">PredictedData</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="whale-docker-images-available">
<span class="emoji" data-emoji="whale">üê≥</span> Docker images available<a class="anchor" aria-label="anchor" href="#whale-docker-images-available"></a>
</h2>
<div class="section level3">
<h3 id="resvidex-as-a-shiny-app-in-docker">
<strong>ReSVidex</strong> as a shiny app in docker<a class="anchor" aria-label="anchor" href="#resvidex-as-a-shiny-app-in-docker"></a>
</h3>
<p>Another way to run <strong>ReSVidex</strong> as a shiny app is to use the docker image. Follow these steps:</p>
<ol style="list-style-type: decimal">
<li><p>If you don¬¥t already have it, install docker: <a href="https://www.docker.com/get-started" class="external-link uri">https://www.docker.com/get-started</a>.</p></li>
<li><p>Open a terminal and run the following:</p></li>
</ol>
<div class="sourceCode" id="cb4"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb4-1"><a href="#cb4-1" tabindex="-1"></a><span class="ex">docker</span> pull cacciabue/resvidex:shiny</span></code></pre></div>
<p>and wait for the image to download. You only have to run this command the first time, or whenever you want to check for updates.</p>
<ol start="3" style="list-style-type: decimal">
<li>Once the downloading is complete run the following:</li>
</ol>
<div class="sourceCode" id="cb5"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb5-1"><a href="#cb5-1" tabindex="-1"></a><span class="ex">docker</span> run <span class="at">-d</span> <span class="at">--rm</span> <span class="at">-p</span> 3838:3838 cacciabue/resvidex:shiny</span></code></pre></div>
<ol start="4" style="list-style-type: decimal">
<li>Finally, open your favorite browser and go to <a href="http://localhost:3838/" class="external-link uri">http://localhost:3838/</a>
</li>
<li>The app should be up and running in your browser. Load the fasta file and press RUN.</li>
<li>You can save a report using the corresponding button.</li>
</ol>
</div>
<div class="section level3">
<h3 id="resvidex-in-a-docker-image-with-all-dependencies-already-installed">
<strong>ReSVidex</strong> in a docker image with all dependencies already installed<a class="anchor" aria-label="anchor" href="#resvidex-in-a-docker-image-with-all-dependencies-already-installed"></a>
</h3>
<p>For more reproducibility a fully operational environment is available to work directly in docker:</p>
<ol style="list-style-type: decimal">
<li><p>If you don¬¥t already have it, install docker: <a href="https://www.docker.com/get-started" class="external-link uri">https://www.docker.com/get-started</a>.</p></li>
<li><p>Open a terminal, go to the directory where the fasta files are stored and run the following:</p></li>
</ol>
<div class="sourceCode" id="cb6"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb6-1"><a href="#cb6-1" tabindex="-1"></a><span class="ex">docker</span> pull cacciabue/resvidex:cli</span></code></pre></div>
<ol start="3" style="list-style-type: decimal">
<li>Once the downloading is complete run the following:</li>
</ol>
<div class="sourceCode" id="cb7"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb7-1"><a href="#cb7-1" tabindex="-1"></a><span class="co">#for WINDOWS </span></span>
<span id="cb7-2"><a href="#cb7-2" tabindex="-1"></a><span class="ex">docker</span> run <span class="at">-it</span> <span class="at">--rm</span> <span class="at">--volume</span> %cd%:/nexus cacciabue/resvidex:cli</span>
<span id="cb7-3"><a href="#cb7-3" tabindex="-1"></a></span>
<span id="cb7-4"><a href="#cb7-4" tabindex="-1"></a><span class="co">#for unix/MAC</span></span>
<span id="cb7-5"><a href="#cb7-5" tabindex="-1"></a><span class="ex">docker</span> run <span class="at">-it</span> <span class="at">--rm</span> <span class="at">--volume</span> <span class="va">$(</span><span class="bu">pwd</span><span class="va">)</span>:/nexus cacciabue/resvidex:cli</span></code></pre></div>
<ol start="4" style="list-style-type: decimal">
<li>Now you can run</li>
</ol>
<div class="sourceCode" id="cb8"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb8-1"><a href="#cb8-1" tabindex="-1"></a><span class="ex">setwd</span><span class="er">(</span><span class="st">'nexus'</span><span class="kw">)</span></span>
<span id="cb8-2"><a href="#cb8-2" tabindex="-1"></a></span>
<span id="cb8-3"><a href="#cb8-3" tabindex="-1"></a><span class="co">#Call the 'resvidex' library</span></span>
<span id="cb8-4"><a href="#cb8-4" tabindex="-1"></a><span class="ex">library</span><span class="er">(</span><span class="ex">resvidex</span><span class="kw">)</span></span>
<span id="cb8-5"><a href="#cb8-5" tabindex="-1"></a></span>
<span id="cb8-6"><a href="#cb8-6" tabindex="-1"></a><span class="co"># Indicate the file path to the fasta file to use.If your file is in your working directory you need to simply indicate the file name. In this case, we use a test file provided with the package itself. </span></span>
<span id="cb8-7"><a href="#cb8-7" tabindex="-1"></a></span>
<span id="cb8-8"><a href="#cb8-8" tabindex="-1"></a><span class="ex">file_path</span><span class="op">&lt;</span>-system.file<span class="er">(</span><span class="st">"extdata"</span><span class="ex">,</span><span class="st">"test_dataset.fasta"</span><span class="ex">,package=</span><span class="st">"resvidex"</span><span class="kw">)</span></span>
<span id="cb8-9"><a href="#cb8-9" tabindex="-1"></a></span>
<span id="cb8-10"><a href="#cb8-10" tabindex="-1"></a><span class="co"># Use the wrapper function. You can pass other arguments</span></span>
<span id="cb8-11"><a href="#cb8-11" tabindex="-1"></a><span class="ex">Classification</span><span class="op">&lt;</span>-classify<span class="er">(</span><span class="va">inputFile</span><span class="op">=</span>file_path,model=FULL_GENOME<span class="kw">)</span></span>
<span id="cb8-12"><a href="#cb8-12" tabindex="-1"></a><span class="ex">Classification</span></span>
<span id="cb8-13"><a href="#cb8-13" tabindex="-1"></a></span>
<span id="cb8-14"><a href="#cb8-14" tabindex="-1"></a><span class="co">#if you want to export the prediction</span></span>
<span id="cb8-15"><a href="#cb8-15" tabindex="-1"></a><span class="ex">utils::write.csv2</span><span class="er">(</span><span class="ex">Classification,</span><span class="st">'Classification_file.csv'</span><span class="kw">)</span></span>
<span id="cb8-16"><a href="#cb8-16" tabindex="-1"></a></span>
<span id="cb8-17"><a href="#cb8-17" tabindex="-1"></a><span class="co"># This command saves a file in the working directory as "Results.csv" by default. You can change the name file setting the "outputFile" parameter.</span></span>
<span id="cb8-18"><a href="#cb8-18" tabindex="-1"></a></span>
<span id="cb8-19"><a href="#cb8-19" tabindex="-1"></a><span class="co"># To exit the container just run</span></span>
<span id="cb8-20"><a href="#cb8-20" tabindex="-1"></a></span>
<span id="cb8-21"><a href="#cb8-21" tabindex="-1"></a><span class="fu">q()</span></span></code></pre></div>
<ol start="5" style="list-style-type: decimal">
<li>Alternatively you can perform step 3 and 4 in one command like this:</li>
</ol>
<div class="sourceCode" id="cb9"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb9-1"><a href="#cb9-1" tabindex="-1"></a></span>
<span id="cb9-2"><a href="#cb9-2" tabindex="-1"></a><span class="co">#for WINDOWS </span></span>
<span id="cb9-3"><a href="#cb9-3" tabindex="-1"></a></span>
<span id="cb9-4"><a href="#cb9-4" tabindex="-1"></a><span class="ex">docker</span> run <span class="at">--rm</span> <span class="at">--volume</span> %cd%:/nexus cacciabue/resvidex:cli R <span class="at">-e</span> <span class="st">"setwd('nexus');library('resvidex');Classification&lt;-classify(inputFile='test_dataset.fasta',model=FULL_GENOME);utils::write.csv2(Classification,'Classification_file.csv')"</span></span>
<span id="cb9-5"><a href="#cb9-5" tabindex="-1"></a></span>
<span id="cb9-6"><a href="#cb9-6" tabindex="-1"></a><span class="co">#for unix/MAC</span></span>
<span id="cb9-7"><a href="#cb9-7" tabindex="-1"></a></span>
<span id="cb9-8"><a href="#cb9-8" tabindex="-1"></a><span class="ex">docker</span> run <span class="at">--rm</span> <span class="at">--volume</span> <span class="va">$(</span><span class="bu">pwd</span><span class="va">)</span>:/nexus cacciabue/resvidex:cli R <span class="at">-e</span> <span class="st">"setwd('nexus');library('resvidex');Classification&lt;-classify(inputFile='test_dataset.fasta',model=FULL_GENOME);utils::write.csv2(Classification,'Classification_file.csv')"</span></span>
<span id="cb9-9"><a href="#cb9-9" tabindex="-1"></a></span>
<span id="cb9-10"><a href="#cb9-10" tabindex="-1"></a></span>
<span id="cb9-11"><a href="#cb9-11" tabindex="-1"></a><span class="co"># USER SHOULD CHANGE test_dataset.fasta for the correct file name</span></span></code></pre></div>
</div>
</div>
</div>

  </main><aside class="col-md-3"><div class="license">
<h2 data-toc-skip>License</h2>
<ul class="list-unstyled">
<li><a href="LICENSE.html">Full license</a></li>
<li><small><a href="https://opensource.org/licenses/mit-license.php" class="external-link">MIT</a> + file <a href="LICENSE-text.html">LICENSE</a></small></li>
</ul>
</div>

<div class="community">
<h2 data-toc-skip>Community</h2>
<ul class="list-unstyled">
<li><a href="CONTRIBUTING.html">Contributing guide</a></li>
<li><a href="CODE_OF_CONDUCT.html">Code of conduct</a></li>
</ul>
</div>

<div class="citation">
<h2 data-toc-skip>Citation</h2>
<ul class="list-unstyled">
<li><a href="authors.html#citation">Citing resvidex</a></li>
</ul>
</div>

<div class="developers">
<h2 data-toc-skip>Developers</h2>
<ul class="list-unstyled">
<li>Marco Cacciabue <br><small class="roles"> Author, maintainer </small> <a href="https://orcid.org/http://orcid.org/0000-0002-1429-4252" target="orcid.widget" aria-label="ORCID" class="external-link"><span class="fab fa-orcid orcid" aria-hidden="true"></span></a> </li>
<li>Melina Obregon <br><small class="roles"> Author </small> <a href="https://orcid.org/http://orcid.org/0009-0000-9741-1890" target="orcid.widget" aria-label="ORCID" class="external-link"><span class="fab fa-orcid orcid" aria-hidden="true"></span></a> </li>
<li>Nahuel Fenoglio <br><small class="roles"> Author </small> <a href="https://orcid.org/http://orcid.org/0009-0007-3163-2181" target="orcid.widget" aria-label="ORCID" class="external-link"><span class="fab fa-orcid orcid" aria-hidden="true"></span></a> </li>
</ul>
</div>

<div class="dev-status">
<h2 data-toc-skip>Dev status</h2>
<ul class="list-unstyled">
<li><a href="https://zenodo.org/badge/latestdoi/COMPLETE!!" class="external-link"><img src="https://zenodo.org/badge/COMPLETE!!" alt="DOI"></a></li>
<li><a href="https://www.tidyverse.org/lifecycle/#experimental" class="external-link"><img src="https://img.shields.io/badge/lifecycle-experimental-orange.svg"></a></li>
</ul>
</div>

  </aside>
</div>


    <footer><div class="pkgdown-footer-left">
  <p></p>
<p>Developed by Marco Cacciabue, Melina Obregon, Nahuel Fenoglio.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
